hcatalog/webhcat/svr/src/main/java/org/apache/hive/hcatalog/templeton/Main.java

```diff
@@ -306,18 +308,30 @@ private Connector createChannelConnector(Server server) {
   public FilterHolder makeAuthFilter() throws IOException {
     FilterHolder authFilter = new FilterHolder(AuthFilter.class);
     UserNameHandler.allowAnonymous(authFilter);
+  
+    // compatible with Hadoop 3.3.x.
```

Comment:
Can you elaborate a bit more what compatible means.

If I understand well the changes are required after upgrading to hadoop 3.3.x onwards otherwise the Authentication filter will not work but what happens with versions before 3.3.x.  I guess that the new code will not work if an earlier version of hadoop is used.

Reply:
If we don't care about versions before 3.3.x, we need to use AuthFilterInitializer, refers to:https://github.com/apache/hadoop/commit/4ea6c2f457496461afc63f38ef4cef3ab0efce49#diff-8794afa0dfd45cc8be6743dd6d13d0bd89bdfbd6dbab8804da0b0555413734d6R67

`AuthFilterInitializer` not exists in earlier version in Hadoop, that's why I set all configuration by myself, I'm trying to compatible with both Hadoop3.1.x and Hadoop3.3.x, But I only tested with Hadoop3.3.x

Reply:
Hive is currently on Hadoop 3.3.1 and this is the only version that we are gonna claim that we are compatible with. You can pick whatever API you think its best as long as it works fine for 3.3.1. 

The comment can be removed since by definition we are compatible **only** with the `hadoop.version` mentioned in the pom files.